src/client/job.rs

```diff
@@ -57,20 +57,39 @@ pub fn job_status(info: &JobInfo) -> Status {
 fn task_status(status: &JobTaskState) -> Status {
     match status {
         JobTaskState::Waiting => Status::Waiting,
-        JobTaskState::Running => Status::Running,
-        JobTaskState::Finished => Status::Finished,
+        JobTaskState::Running(_) => Status::Running,
+        JobTaskState::Finished(_) => Status::Finished,
         JobTaskState::Failed(_) => Status::Failed,
         JobTaskState::Canceled => Status::Canceled,
     }
 }
 
-fn status_cell(status: Status) -> CellStruct {
-    match status {
-        Status::Waiting => "WAITING".cell().foreground_color(Some(Color::Cyan)),
-        Status::Finished => "FINISHED".cell().foreground_color(Some(Color::Green)),
-        Status::Failed => "FAILED".cell().foreground_color(Some(Color::Red)),
-        Status::Running => "RUNNING".cell().foreground_color(Some(Color::Yellow)),
-        Status::Canceled => "CANCELED".cell().foreground_color(Some(Color::Magenta)),
+fn worker_state_messages(message: &str, worker_id: &WorkerId, color: Color) -> CellStruct {
+    let mut message = String::from(message);
+    message.push_str(worker_id.to_string().as_str());
+    message.cell().foreground_color(Some(color))
+}
+
+fn status_cell(status: Status, task_state: Option<&JobTaskState>) -> CellStruct {
+    match task_state {
+        Some(state) => match state {
+            JobTaskState::Waiting => "WAITING".cell().foreground_color(Some(Color::Cyan)),
+            JobTaskState::Finished(worker_id) => {
+                worker_state_messages("FINNISHED ON ", worker_id, Color::Green)
```

Comment:
You can just use the `format!` macro.

`format!("FINISHED ON {}", worker_id)`

Also I would switch the matches (first match `state`, then inside choose specific message based on task_state, to avoid duplicating the colors.